ui/models/push.js

```diff
@@ -151,24 +154,44 @@ export default class PushModel {
   }
 
   static async getDecisionTaskId(pushId, notify) {
-    const taskIdMap = await PushModel.getDecisionTaskMap([pushId], notify);
+    const taskId = decisionTaskIdCache[pushId];
 
-    return taskIdMap[pushId];
+    if (taskId) {
+      return taskId;
+    }
+    Object.assign(
+      decisionTaskIdCache,
+      await PushModel.getDecisionTaskMap([pushId], notify),
+    );
+    return decisionTaskIdCache[pushId];
   }
 
   static async getDecisionTaskMap(pushIds, notify) {
+    // If any Push ids are not yet in the cache, then fetch them.
+    // Otherwise just return the map since it has everything
+    // that's needed.
+    const cachedMap = pick(decisionTaskIdCache, pushIds);
+    const missedIds = pushIds.filter(id => !cachedMap[id]);
+
+    if (!Object.keys(missedIds).length) {
```

Comment:
It looks like `missedIds` will be an array - empty or not - not an object.

Reply:
Sheesh... right you are!  Nice catch.  :)